src/kernels/MIOpenKthvalue.cpp

```diff
@@ -195,3 +195,60 @@ extern "C" __global__ void KthvalueFwd(const IN_OUT_TYPE* input,
                              output_tv,
                              indices_tv);
 }
+
+template <typename DTYPE>
+__device__ void kthvalue_bwd(DTYPE* input_grad,
+                             const DTYPE* output_grad,
+                             const uint64_t* indices,
+                             uint64_t dim_size,
+                             uint64_t dim_stride,
+                             tensor_view_t<4> input_grad_tv,
+                             tensor_view_t<5> output_grad_tv,
+                             tensor_view_t<5> indices_tv)
```

Comment:
See @hieule88 [kernels code](https://github.com/hieule88/MIOpen/blob/e5bc98c4ea6a5a9cb9733c014ab39b16e90eb272/src/kernels/MIOpenUnsortedSegmentSum.cpp#L35) to know how to handle multiple datatype of `indices`.

Reply:
Please refer to my [comment](https://github.com/moreh-dev/MIOpen/pull/71#discussion_r1897151172) above, where I explain why I think supporting multiple data types for indices is unnecessary